RUFAS/input_manager.py

```diff
@@ -2028,6 +2044,271 @@ def dump_get_data_logs(self, path: Path) -> None:
+            "function": self._validate_metadata.__name__,
+        }
+        metadata_files = self.__metadata["files"]
+        required_keys = {"path", "type", "properties"}
+        optional_keys = {"title", "description"}
+        valid_keys = required_keys | optional_keys
+        valid_types = {"json", "csv"}
+        for key, data in metadata_files.items():
+            if not required_keys <= data.keys():
+                missing_keys = required_keys - data.keys()
+                om.add_error("Metadata Validation", f"Missing required keys '{missing_keys}' in {key}", info_map)
+                raise ValueError
+            invalid_keys = data.keys() - valid_keys
+            if invalid_keys:
+                om.add_error("Metadata Validation", f"Invalid keys '{invalid_keys}' in {key}", info_map)
+                raise ValueError
+
+            if data["type"] not in valid_types:
+                om.add_error(
+                    "Metadata Validation",
+                    f"Invalid type '{data['type']}' in {key}. Expected 'json' or 'csv'.",
+                    info_map,
+                )
+                raise ValueError
+
+            if not os.path.isfile(data["path"]):
+                om.add_error("Metadata Validation", f"Invalid path '{data['path']}' in {key}", info_map)
+                raise ValueError
+
+        om.add_log("Metadata Validation", "Top level metadata is valid.", info_map)
+
+    def _validate_properties(self) -> None:
+        """Iteratively traverses the metadata properties to check the max depth and routes
+        primitive properties to be validated by type.
+
+        Raises
+        ------
+        ValueError
+            If the depth of the metadata exceeds the metadata_depth_limit.
+        """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._validate_properties.__name__,
+        }
+
+        stack = [(self.__metadata["properties"], 0, [])]
+        max_depth = 0
+        deepest_path = []
+
+        type_to_validator_map: Dict[str, Callable] = {
+            "number": self._metadata_number_validator,
+            "array": self._metadata_array_validator,
+            "bool": self._metadata_bool_validator,
+            "string": self._metadata_string_validator,
+        }
+        while stack:
+            current_obj, depth, path = stack.pop()
+
+            if depth > self.metadata_depth_limit:
+                om.add_error(
+                    "Max metadata depth exceeded.",
+                    "Metadata depth exceeds maximum allowed depth of " f"{self.metadata_depth_limit} at path {path}",
+                    info_map,
+                )
+                raise ValueError(
+                    "Metadata depth exceeds maximum allowed depth of " f"{self.metadata_depth_limit} at path {path}"
+                )
+
+            if depth > max_depth:
+                max_depth = depth
+                deepest_path = path
```

Comment:
```suggestion
        current_max_depth = 0
        deepest_path = []

        type_to_validator_map: Dict[str, Callable] = {
            "number": self._metadata_number_validator,
            "array": self._metadata_array_validator,
            "bool": self._metadata_bool_validator,
            "string": self._metadata_string_validator,
        }
        while stack:
            current_obj, depth, path = stack.pop()

            if depth > self.metadata_depth_limit:
                om.add_error(
                    "Max metadata depth exceeded.",
                    "Metadata depth exceeds maximum allowed depth of " f"{self.metadata_depth_limit} at path {path}",
                    info_map,
                )
                raise ValueError(
                    "Metadata depth exceeds maximum allowed depth of " f"{self.metadata_depth_limit} at path {path}"
                )

            if depth > current_max_depth:
                current_max_depth = depth
                deepest_path = path
```
maybe rename this local variable to avoid confusion with `self.metadata_depth_limit`